### ISSUE TITLE
Adding support for _checked state selector to Checkbox container

### ISSUE THREAD

**Issue by codetheweb** [chakra-ui/chakra-ui#4647]
### Description

It can be useful to be able to set the container style of a <Checkbox/> depending on whether or not it's checked. For example, changing the background to make the checked state stand out.

### Problem Statement/Justification

Some design systems may require being able to change the container styling depending on the checked state.

Also makes sense to add for a more consistent API as both the `control` and `label` parts seem to support a `_checked` state.

### Proposed Solution or API

Something like 

```js
{
  baseStyle: {
    container: {
      _checked: {
        bg: 'red.200'
      }
    }
  }
}
```

### Alternatives

Currently making a custom <Checkbox/> component.

### Additional Information

Example:

![Screen Shot 2021-08-25 at 2 12 19 PM](https://user-images.githubusercontent.com/7410405/130850955-60a06157-6234-4dd2-86e1-4455e6b089e0.png)

**Comment by codetheweb** [comment]
Right, that works for the `control` part. I need to style the `container` based on the checked state (see screenshot).

**Comment by anubra266** [comment]
Oh...but the Checkbox doesn't have a container part

**Comment by codetheweb** [comment]
It did ~10 days ago when I was working on this. 😛 Looks like stuff's been updated since then. Although the Checkbox component [still references a `container` part](https://github.com/chakra-ui/chakra-ui/blob/a8bfa9f5fdaa840a9035024fc9e434b32244c83f/packages/checkbox/src/checkbox.tsx#L149)?

Regardless, my question about styling the container (or whatever else it's now / will be called) based on the checked state still stands.

**Comment by segunadebayo** [comment]
@codetheweb Yes, it still stands and I think it's a valid use-case.

I've just added support for this. It'll be available in the next release.

**Comment by codetheweb** [comment]
Great, thank you!
